clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -567,6 +572,10 @@ public void handle(JoinGroupResponse joinResponse, RequestFuture<ByteBuffer> fut
                     AbstractCoordinator.this.state = MemberState.UNJOINED;
                 }
                 future.raise(Errors.MEMBER_ID_REQUIRED);
+            } else if (error == Errors.GROUP_INSTANCE_ID_NOT_FOUND) {
```

Comment:
Maybe unknown instance id is more aligned? It's slightly different comparing with member id unknown.

Reply:
Our response to this error is to discard our current memberId and rejoin. That seems true regardless whether static or dynamic membership is used, so I thought we may as well make the error consistent. Does that make sense?

Reply:
Although we would react the error with same handling logic, I do see the benefit of decoupling error for now, because the error log could better help user triage during consumer incident.

Reply:
I see your point Jason, make sense here. The logic is the since `GROUP_INSTANCE_ID_NOT_FOUND` is not covering the whole cases (like when member id is unknown), we could just bypass this check.